src/providers/eth_provider/starknet/relayer.rs

```diff
@@ -24,33 +23,35 @@ static RELAYER_SIGNER: LazyLock<LocalWallet> = LazyLock::new(|| {
     ))
 });
 
-/// A relayer holding a lock on a mutex on an account and connected to the Starknet network.
+/// A relayer holding an account and a balance.
+///
 /// The relayer is used to sign  transactions and broadcast them on the network.
 #[derive(Debug)]
-pub struct LockedRelayer<'a, SP: Provider + Send + Sync> {
+pub struct Relayer<SP: Provider + Send + Sync> {
     /// The account used to sign and broadcast the transaction
     account: SingleOwnerAccount<SP, LocalWallet>,
     /// The balance of the relayer
     balance: Felt,
-    /// The locked nonce held by the relayer
-    nonce: MutexGuard<'a, Felt>,
 }
 
-impl<'a, SP> LockedRelayer<'a, SP>
+impl<SP> Relayer<SP>
 where
     SP: Provider + Send + Sync,
 {
-    /// Create a new relayer with the provided Starknet provider, address, balance and nonce.
-    pub fn new(lock: MutexGuard<'a, Felt>, address: Felt, balance: Felt, provider: SP, chain_id: Felt) -> Self {
+    /// Create a new relayer with the provided Starknet provider, address, balance.
+    pub fn new(address: Felt, balance: Felt, provider: SP, chain_id: Felt) -> Self {
```

Comment:
here let's remove the chain id and replace it with `STARKNET_CHAIN_ID`

Reply:
yep fixed